src/olympia/devhub/utils.py

```diff
@@ -307,3 +312,50 @@ def add_dynamic_theme_tag(version):
     files = version.all_files
     if any('theme' in file_.webext_permissions_list for file_ in files):
         Tag(tag_text='dynamic theme').save_tag(version.addon)
+
+
+def get_addon_akismet_reports(user, user_agent, referrer, upload=None,
+                              addon=None, data=None):
+    if not waffle.switch_is_active('akismet-spam-check'):
+        return []
+    assert addon or upload
+    properties = ('name', 'summary', 'description')
+
+    if upload:
+        addon = addon or upload.addon
+        data = data or Addon.resolve_webext_translations(
+            parse_addon(upload, addon, user), upload)
+
+    if not data:
+        return []  # bail early if no data to skip Translation lookups
+    if addon and addon.has_listed_versions():
+        translation_ids = (
+            getattr(addon, prop + '_id', None) for prop in properties)
+        translation_ids = (id_ for id_ in translation_ids if id_)
```

Comment:
Better safe than sorry, let's make this a list instead of a generator, Django will convert it anyhow right away